CAIPs/caip-211.md

```diff
@@ -0,0 +1,275 @@
+to layer custom or local RPC semantics and/or routing onto a session. Since
+CAIP-25 respondents ignore unknown properties, respondents that conform to
+CAIP-25 but not to CAIP-211 should be carefully considered and accomodated by
+implementers. 
+
+## Motivation
+
+While some core methods and notifications are foundational to entire namespaces
+and thus almost universally defined out-of-band (meaning all callers and
+respondents agree to them already), others are specific to chains or even to
+subsets of wallets and dapps on a given chain.  This requires scope objects in
+CAIP-25 requests to negotation semantic anchors and/or network routing
+**before** authorization can occur in the confidence that both parties agree to
+the syntax and semantics of a given method or notification, which may only be
+available on certain RPC endpoints.
+
+The concept of a custom RPC endpoint or definition presumes the existence of
+"default" or universal endpoints and definitions, which can be hard to anchor in
+explicit specifications unless a namespace profile of this CAIP has been
+published. (This may be unnecessary in the case of younger namespaces without
+customization built in at the per-chain or per-dapp layer). An empty array of
+`rpcEndpoints` or `rpcDocuments` sent in either direction should be interpreted
+differently than the absence of the array.  Namely, either array being present
+but empty in a request signals that a caller is requesting that custom endpoints
+or  definitions NOT be considered in CAIP-25 authorizations. Present but empty
+arrays in a response confirms this behavior, while its absence can be taken to
+mean the wallet either does not support custom endpoints/definitions or the user
+has opted out of that degree of trust, which for privacy/fingerprinting reasons,
+should not be handled by distinct codepaths (See Privacy Considerations).
+
+## Specification
+
+Two properties are added to the scope objects requested in `optionalScopes` and
+optionally returned in `sessionScopes`.  These are both [ordered] strings of arrays:
+
+1. `rpcEndpoints` is an array of zero or more URLs of RPC endpoints that
+   the caller would prefer the respondent to use, ordered by preference. Each
+   must be a valid URL that addresses an RPC endpoint. The respondent may
+   return it empty, reordered, with less, the same, or even more conformant URLs
+   than received.
+2. `rpcDocuments` is an array of zero or more URLs of machine-readable RPC
+   documents that the caller would prefer the respondent to use, ordered by
+   preference. This set of document collectively defines at least syntactically
+   if not also semantically any methods and/or notifications authorized by the
+   CAIP-25 authorization, in DESCENDING heirarchical authority. (For example,
+   any methods or notifications defined differently by multiple authorites will
+   be interpreted by whichever authority is closer to the 0-index of the array).
```

Comment:
This is not entirely clear to me, is it possible to re-word this?

Reply:
reworded in [newest commit](https://github.com/ChainAgnostic/CAIPs/pull/211/commits/6748a7316ce6615f5bdd22a499819b8dde4b3f0e)